src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,649 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.permissions.ForgePermissionNodes;
+
+import net.minecraft.command.CommandBase;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Command entry point to make minecraft inheritance happy.
+ */
+public abstract class AbstractCommandParser extends CommandBase
+{
+    public interface PermissionsChecker
+    {
+        boolean hasPermission(ForgePermissionNodes forgePermissionNode, EntityPlayer player);
+        boolean canUseCommands(MinecraftServer server, ICommandSender sender);
+    }
+
+    public interface ModuleContext {
+        /**
+         * @return the object that is known in this context for this type.
+         */
+        <T> T get(Class<? extends T> type);
+    }
+    
+    /**
+     * Check permissions for executing commands
+     */
+    @NotNull
+    abstract protected PermissionsChecker getPermissionsChecker();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)